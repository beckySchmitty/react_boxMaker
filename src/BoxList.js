import React, {useState } from "react";
import Box from "./Box"
import { v4 as uuidv4 } from 'uuid';


const BoxList = () => {

    const initialState = [
        {id: uuidv4(),height: '200', width: '300', color: 'red' },
        { id: uuidv4(), height: '200', width: '300', color: 'blue' }
    ]
    const [boxes, setBoxes] = useState(initialState);

    // const handleSubmit = () => {
    //     setBoxes([{height: '200px;', width: '300px;', color: 'red;' },{height: '200px;', width: '300px;', color: 'blue;' }])
    // }

    const handleClick = () => {
        setBoxes([
            {id: uuidv4(), height: '200', width: '300', color: 'red' },
            { id: uuidv4(), height: '200', width: '300', color: 'blue' }
        ])
    }

    return (
        <div>
            {boxes.map(box => <Box key={box.id} color={box.color} width={box.width} height={box.height} handleClick={handleClick}/>)}
        </div>
    )
}

export default BoxList;