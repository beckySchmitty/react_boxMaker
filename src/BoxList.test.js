import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", function () {
  render(<BoxList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add new box", function () {
    const boxList = render(<BoxList />);
    // no boxes yet
    expect(boxList.queryByText("X")).not.toBeInTheDocument();
    // add box 
    const heightInput = boxList.getByLabelText("Height");
    const widthInput = boxList.getByLabelText("Width");
    const backgroundInput = boxList.getByLabelText("Color");
    fireEvent.change(backgroundInput, { target: { value: "blue" } });
    fireEvent.change(widthInput, { target: { value: '500' } });
    fireEvent.change(heightInput, { target: { value: '500' } });
    const button = boxList.getByText("Add Box!");
    fireEvent.click(button);

    const removeBtn = boxList.getByText("X");
    expect(removeBtn).toBeInTheDocument();
    expect(removeBtn.previousSibling).toHaveStyle(`
    width: 500px;
    height: 500px;
    background-color: blue;
  `);

})
