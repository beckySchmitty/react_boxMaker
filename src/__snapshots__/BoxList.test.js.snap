// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <label
          for="height"
        >
          Height
        </label>
        <input
          id="height"
          name="height"
          placeholder="height"
          type="text"
          value=""
        />
        <label
          for="width"
        >
          Width
        </label>
        <input
          id="width"
          name="width"
          placeholder="width"
          type="text"
          value=""
        />
        <label
          for="color"
        >
          Color
        </label>
        <input
          id="color"
          name="color"
          placeholder="color"
          type="text"
          value=""
        />
        <button>
          Add Box!
        </button>
      </form>
    </div>
    <div
      style="height: 200px; width: 300px; background-color: red;"
    >
      <button>
        X
      </button>
    </div>
    <div
      style="height: 200px; width: 300px; background-color: blue;"
    >
      <button>
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
